src/GafferSceneUI/VisualiserTool.cpp

```diff
@@ -762,13 +762,33 @@ class VisualiserGadget : public Gadget
 				ScenePlug::PathScope scope( &location.context(), &location.path() );
 
 				// Check path exists
-				if( !location.scene().existsPlug()->getValue() )
+				try
 				{
+					if( !location.scene().existsPlug()->getValue() )
+					{
+						continue;
+					}
+				}
+				catch( const std::exception & )
+				{
+					/// \todo Ideally the GL state would be handled by `IECoreGL::State` and related classes
+					/// which would restore the GL state via RAII in the case of exceptions.
+					/// But those don't handle everything we need like shader attribute block alignment,
+					/// `GL_POLYGON_OFFSET` and more, so we use try / catch blocks throughout this tool.
 					continue;
 				}
 
 				// Extract mesh primitive
-				auto mesh = runTimeCast<const MeshPrimitive>( location.scene().objectPlug()->getValue() );
+				ConstMeshPrimitivePtr mesh;
+				try
+				{
+					mesh = runTimeCast<const MeshPrimitive>( location.scene().objectPlug()->getValue() );
+				}
+				catch( const std::exception & )
+				{
+					continue;
+				}
```

Comment:
It feels like this might be clearer as a single try/catch block with all the scene accesses in it (including the transform access lower down). What do you think?

Reply:
That looks good to me, I grouped those in https://github.com/GafferHQ/gaffer/pull/6347/commits/3ff4bbac0ed10f819f73a5ec565968b8fc95faa8.